src/blog/2024/01/send-a-file.md

```diff
@@ -0,0 +1,166 @@
+
+### Instructions ###
+1. Install Node-RED Dashboard 2.0.  Follow these [instructions](https://dashboard.flowfuse.com/getting-started.html) to install.  
+2. Import Flow - to import the flow into your Node-RED instance follow these [instructions](https://flowfuse.com/blog/2023/03/3-quick-node-red-tips-5/#1.-copy-and-share-your-flows-using-export-and-import). 
+3. Access Dashboard - To access the dashboard, navigate to the ```http://hostname:1880/dashboard```.
+
+This dashboard is currently configured to take in CSV files and transform them into a single message that is sent to the table for visualization.  Simultaneously the data from the import is stored locally in the flow context.  From there, the button can be used to trigger the sending of the data from the flow context to the next destination.  In this case, it is a simple debug node.
+
+
+<iframe width="100%" height="225px" src="https://flows.nodered.org/flow/8c505039ac1b8dbed2bee1e22ee2975a/share?height=100" allow="clipboard-read; clipboard-write" style="border: none;"></iframe>
+
+
+## Simple Python Script
+
+This simple Python script sends a file to a Node-RED flow.  The flow that will work with can be seen [here](#node-red-ingress).
+
+The script requires **requests** and **Python 3.x**.
+
+Install requests:
+
+```bash
+pip install requests
+```
+
+Create a file called run.py and paste the contents into the file.
+
+```python
+import requests
+
+def send_file(nodered_url, file_path):
+    # Open the file in binary mode
+    with open(file_path, 'rb') as file:
+        files = {'file': (file.name, file, 'multipart/form-data')}
+        response = requests.post(nodered_url, files=files)
+
+    return response
+
+# Update the ip address and port of your Node-RED instance
+nodered_url = 'http://localhost:1880/fileupload'
+
+# Update the location of your file
+file_path = 'C:/Users/myUser/Downloads/shiftSchedule.csv'
+
+response = send_file(nodered_url, file_path)
+print(f"Response Status Code: {response.status_code}")
+print(f"Response Body: {response.text}")
+```
+
+Update the **nodered_url** to the location of the NodeRED instance.  Be sure to adjust the port if the default port of 1880 isn't being used.
+
+Update the **file_path** with the path to where the file to be uploaded will be located.
+
+**Save**
+
+To run:
+
+```python
+python run.py
+```
+
+
+## Stand Alone Web Application
+
+![csv upload application](./images/csv_upload_app.png)
+
+This stand-alone web application can be run on either Windows or Linux, .bat for Windows, and .sh for Linux.
+
+### Installation
+
+Clone the repository:
+```bash
+git clone https://github.com/gdziuba/FF_Send-File-to-NR.git
+```
+Navigate to the directory:
+```bash
+cd FF_Send-File-to-NR
+```
```

Comment:
```suggestion
Clone the repository and navigate to the directory:
```bash
git clone https://github.com/gdziuba/FF_Send-File-to-NR.git && cd FF_Send-File-to-NR
```
```